File: python_demo/consumer.py
# Python-based Apache Kafka consumer

from kafka import KafkaConsumer

consumer = KafkaConsumer(
    bootstrap_servers=['localhost:9092', 'localhost:9093', 'localhost:9094'],
    client_id='python-consumer',
    group_id='python-group',
    #  key_deserializer=bytes.decode,
    value_deserializer=bytes.decode
)

consumer.subscribe('names')

for msg in consumer:
    string = (f'partition: {msg.partition}, offset: {msg.offset}, '
              f'value: {msg.value}')
    print(string)

consumer.close()

File: javascript_demo/consumer.js
'use strict';

const { Kafka } = require('kafkajs');

const kafka = new Kafka({
  clientId: 'my-consumer',
  brokers: ['localhost:9092', 'localhost:9093', 'localhost:9094'],
});

const consumer = kafka.consumer({
  groupId: 'first-group',
});

const runConsumer = async (args) => {
  if (args.length >= 1) {
    let topic = args[0];

    await consumer.connect();
    await consumer.subscribe({
      topic,
      fromBeginning: false,
    });

    await consumer.run({
      eachMessage: async ({ topic, partition, message }) => {
        console.log({
          partition: partition,
          offset: message.offset,
          value: message.value.toString(),
        });
      },
    });
  } else {
    console.log('First argument to consumer must be the topic');
  }
};

module.exports = { runConsumer };

const args = process.argv.slice(2);
runConsumer(args).catch(console.error);
